### PR TITLE
CLN: explicit kwargs for select

### PR DESCRIPTION
**Pull request by jbrockmendel**
Questions for @jreback on follow-ups:

1) Note in `read_hdf` kwargs is still present in the signature, but i put an assertion that they are empty.  In principle we could pass more junk to HDFStore, but we never do in tests.  Should we allow it, or can those kwargs be deleted altogether?

2) the current signature for `select_column` is `(self, key: str, column: str, **kwargs)`.  These kwargs are passed to `read_column`, which would accept "where", "start", "stop".  In tests we pass start and stop, but never where.  Can we rule out "where"?

### CODE REVIEW

**Comment by jreback** [comment]
> Questions for @jreback on follow-ups:
> 
> 1. Note in `read_hdf` kwargs is still present in the signature, but i put an assertion that they are empty.  In principle we could pass more junk to HDFStore, but we never do in tests.  Should we allow it, or can those kwargs be deleted altogether?
> 2. the current signature for `select_column` is `(self, key: str, column: str, **kwargs)`.  These kwargs are passed to `read_column`, which would accept "where", "start", "stop".  In tests we pass start and stop, but never where.  Can we rule out "where"?

IIRC I answered 1) elsewhere. we pass on kwargs in the constructor to ``.open_file``; these are keywords that are used in the construction deep in PyTables for how to open the backends, e.g.https://www.pytables.org/cookbook/inmemory_hdf5_files.html, e.g. you could pass ``driver=`` and it would work.

2) yes I think you can remove ``where``

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -284,7 +284,19 @@ def to_hdf(
...
+    key=None,
+    mode: str = "r",
+    errors: str = "strict",
+    where=None,
+    start: Optional[int] = None,
+    stop: Optional[int] = None,
+    columns=None,
+    iterator=False,
+    chunksize: Optional[int] = None,
+    **kwargs,
```
Comment (jreback) [inline pandas/io/pytables.py]: a solution to this would be to remove kwargs and instead add a keyword ``backend_kwargs={}`` or ``driver_kwargs={}``

**Comment by jreback** [comment]
suggestion for a way out to remove the kwargs arg in read_hdf & HDFStore

File: pandas/io/pytables.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: that would work, will need deprecation cycle, and im all about _removing_ deprecations these days

Comment (jbrockmendel) [reply pandas/io/pytables.py]: for now ill need to make a quick PR reverting the `assert not kwargs` this introduced

**Pull request merged**